src/main/java/teammates/logic/api/Logic.java

```diff
@@ -361,13 +361,13 @@ public InstructorAttributes updateInstructor(InstructorAttributes.UpdateOptionsW
      * Preconditions: <br>
      * * All parameters are non-null.
      */
-    public InstructorAttributes joinCourseForInstructor(String encryptedKey, String googleId, String institute)
+    public InstructorAttributes joinCourseForInstructor(String encryptedKey, String googleId, String institute, String mac)
             throws InvalidParametersException, EntityDoesNotExistException, EntityAlreadyExistsException {
 
         Assumption.assertNotNull(googleId);
         Assumption.assertNotNull(encryptedKey);
 
```

Comment:
The problem is that the institute can be null? From what I gather reading the method. If the institute is null, an institute gets assigned from the course the instructor is joining. It's weird but it's what's already there :")

Reply:
Ahh, ok. Let's update the docs then, since it currently says "All parameters are non-null."

Reply:
I was actually thinking maybe it could be the other way around? We have the chance now to edit this method and ensure that all parameters are non-null :)

Reply:
Sorry I was referring to the method with the same name in `AccountsLogic.java`